test/python/http_recorder.py

```diff
@@ -0,0 +1,280 @@
+            serializer (callable): the serializer for dumping cassette_dict in cassette_path.
+        """
+        ids2remove = {'api/users/loginWithToken': ['id',
+                                                   'userId',
+                                                   'created'],
+                      'api/Jobs': ['id',
+                                   'userId',
+                                   'creationDate',
+                                   'qasms.executionId',
+                                   'qasms.result.date',
+                                   'qasms.result.data.time',
+                                   'qasms.result.data.additionalData.seed'],
+                      'api/Backends/ibmqx5/queue/status': ['lengthQueue'],
+                      'api/Backends/ibmqx4/queue/status': ['lengthQueue']}
+        IdRemoverPersister.remove_ids(ids2remove, cassette_dict)
+        super(IdRemoverPersister, IdRemoverPersister).save_cassette(cassette_path,
+                                                                    cassette_dict,
+                                                                    serializer)
+
+
+def http_recorder(vcr_mode):
+    """
+    Creates a VCR object in vcr_mode mode.
+
+    Args:
+        vcr_mode (string): the parameter for record_mode.
+
+    Returns:
+        VCR: a VCR object.
+    """
+    my_vcr = VCR(
+        cassette_library_dir='test/cassettes',
+        record_mode=vcr_mode,
+        match_on=['method', 'scheme', 'host', 'port', 'path', 'unordered_query'],
+        filter_headers=['x-qx-client-application', 'User-Agent'],
+        filter_query_parameters=[('access_token', 'dummyapiusersloginWithTokenid01')],
+        filter_post_data_parameters=[('apiToken', 'apiToken_dummy')],
+        decode_compressed_response=True,
+        before_record_response=_purge_headers(['Date',
+                                               ('Set-Cookie', 'dummy_cookie'),
+                                               'X-Global-Transaction-ID',
+                                               'Etag',
+                                               'Content-Security-Policy',
+                                               'X-Content-Security-Policy',
+                                               'X-Webkit-Csp',
+                                               'content-length']))
+    my_vcr.register_matcher('unordered_query', _unordered_query_matcher)
+    my_vcr.register_persister(IdRemoverPersister)
+    return my_vcr
+
+
+def _purge_headers(headers):
+    """
+    Remove headers from the response.
+
+    Args:
+        headers (list): headers to remove from the response
+
+    Returns:
+        callable: for been used in before_record_response VCR constructor.
+    """
+
+    header_list = []
+    for item in headers:
+        if not isinstance(item, tuple):
+            item = (item, None)
+        header_list.append((item[0], item[1]))
+
+    def before_record_response(response):
```

Comment:
Can you add the suffix `_callback` or `_cb` to make it clear this is a callback?

Reply:
b0b1cc6